ui/src/app/pages/admin/admin-user-profile.tsx

```diff
@@ -468,11 +468,11 @@ export const AdminUserProfile = (spinnerProps: WithSpinnerOverlayProps) => {
   };
 
   const updateContactEmail = async (contactEmail: string) => {
+    updateProfile({ contactEmail });
```

Comment:
This was the problem.  It waited for an API call to return before then updated state.  This caused delay and re-rendering.

Now it updates the email state immediately, for well behaved UX